config/indexer.go

```diff
@@ -21,14 +21,21 @@ type Indexer struct {
 	// ShutdownTimeout is the duration that a graceful shutdown has to complete
 	// before the daemon process is terminated.
 	ShutdownTimeout Duration
-	// Directory where value store is kept. If this is not an absolute path
+	// ValueStoreDir is the directory where value store is kept. If this is not an absolute path
 	// then the location is relative to the indexer repo directory.
 	ValueStoreDir string
-	// Type of valuestore to use, such as "sth" or "pogreb".
+	// ValueStoreType specifies type of valuestore to use, such as "sth" or "pogreb".
 	ValueStoreType string
-	// bits for bucket size in store the hash. Note: this should not be changed
-	// from it's value at initialization or the datastore will be corrupted.
+	// STHBits is bits for bucket size in store the hash. Note: this should not be changed
+	// from its value at initialization or the datastore will be corrupted.
 	STHBits uint8
+	// ValueStoreSerde configures the marshalling format of values stored by the valuestore.
+	// It can be one of "json" or "binary". If unspecified, json format is used by default.
```

Comment:
If left unspecified, could the functionality instead be to use whatever the existing value store uses?  If there is no existing value store, then use binary by default.

Reply:
That needs extra engineering in go-indexer-core to probe the existing serialisation format when the file exists

Reply:
Just aksing to consider if that is worth it.

Reply:
Thank you for pointing that out. Left a todo in code.